data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaSinkProducer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.producer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSinkConfig;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+
+import java.util.List;
+
+
+/**
+ * * A helper class which helps takes the buffer data
+ * and produce it to a given kafka topic
+ */
+
+@SuppressWarnings("deprecation")
+public class KafkaSinkProducer {
+
+    final KafkaProducer<String, Object> producer;
+
+    final KafkaSinkConfig kafkaSinkConfig;
+
+    final Record<Event> record;
+
+    final String schemaType;
+
+    public KafkaSinkProducer(final KafkaProducer producer,
+                             final Record<Event> record,
```

Comment:
Not sure why record is part of producer constructor?

Reply:
This has been addressed in PR #2950